digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/AssetFiller.scala

```diff
@@ -779,9 +779,12 @@ class AssetFiller {
     def fillBySideCode(assets: Seq[PieceWiseLinearAsset], roadLink: RoadLinkForFillTopology, changeSet: ChangeSet): (Seq[PieceWiseLinearAsset], ChangeSet) = {
       if (assets.size > 1) {
         val left = assets.head
-        val right = assets.find(sl => sl.startMeasure >= left.endMeasure)
-        if (right.nonEmpty && Math.abs(left.endMeasure - right.get.startMeasure) < MinAllowedLength &&
-          Math.abs(left.endMeasure - right.get.startMeasure) >= Epsilon) {
+        val right = assets.find(sl => sl.startMeasure >= left.endMeasure && sl.sideCode.value == left.sideCode.value)
+        val notTooShortGap = right.nonEmpty && Math.abs(left.endMeasure - right.get.startMeasure) >= Epsilon
+        val gapIsSmallerThanTolerance = right.nonEmpty && Math.abs(left.endMeasure - right.get.startMeasure) < MinAllowedLength
+        val valuesAreSame = right.nonEmpty && left.value.equals(right.get.value)
+        val condition = if (!roadLinkLongAssets.contains(assets.head.typeId)) gapIsSmallerThanTolerance else valuesAreSame
```

Comment:
minkäs takia pitää muuten olla samat arvot?

Reply:
Näin operaattori halusi.

Reply:
Varmaan voi dailyssa viellä miettiä.

Reply:
jos otetaan toi valuesAreSame pois, vähän mietin tuleekos liikaa väärää tietoa, joka olis parempi jättää operaatorin manuaaliseen korjaukseen.

Reply:
Joo, eli operaattori korjaisi sitten tuon InvalidRows csv pohjalta kohteita? Työlistaa ei tainnut olla tälle

Reply:
Varmaan alkuun. Voi siitä tehdä oikean työlistan myöhemmin.